### PR TITLE
desktop app prototype - 0

### PR DESCRIPTION
**Pull request by mrinalwadhwa**
To try the app:

```
cd implementations/typescript
pnpm install

cd implementations/rust/ockam/ockam_app
cargo tauri dev
```

---

We have to resolve with changes in #4895 
[4c5a232](https://github.com/build-trust/ockam/pull/5222/commits/4c5a232beb38958f5371a270dfdcf4be717be13c) is maybe the only changeset here that is not in #4895

### CODE REVIEW

File: tools/nix/parts/all.nix [hunk 1]
```diff
@@ -22,8 +22,11 @@
 
       shellHook = ''
         ${config.devShells.rust.shellHook or ""}
-        export HOME=$PWD/.home
         export DYLD_FALLBACK_LIBRARY_PATH=$(rustc --print sysroot)/lib
+
+        export HOME=$PWD/.home
+        export PATH=$PWD/.home/.cargo/bin:$PATH
+        cargo install tauri-cli --version "^2.0.0-alpha"
```
Comment (shanesveller) [inline tools/nix/parts/all.nix:29]: Later today, I'll contribute an actual Nix derivation for this CLI, so please omit this line and the one prepending PATH prior to merging. 

Generally `shellHook` should not be used to make any permanent modifications to an end user's system, as it violates the principle of least surprise. In a Nix environment there's typically a more idiomatic way to accomplish what's needed.

File: tools/nix/parts/all.nix [hunk 2]
```diff
@@ -22,8 +22,11 @@
 
       shellHook = ''
         ${config.devShells.rust.shellHook or ""}
-        export HOME=$PWD/.home
         export DYLD_FALLBACK_LIBRARY_PATH=$(rustc --print sysroot)/lib
+
+        export HOME=$PWD/.home
```
Comment (shanesveller) [inline tools/nix/parts/all.nix:27]: Please drop this line altogether as well - anyone who uses direnv and nix-direnv (as I do) will find that nearly all of the software on their system that uses XDG config dirs is now broken while inside this devShell. Most values of `$EDITOR`, for example.

@mrinalwadhwa do you recall what this was being set for? Typically we can relocate this env var to a wrapper script or other more localized definition to control for whatever it helps with.

File: tools/nix/parts/all.nix (hunk 1 above)
Comment (etorreborre) [reply tools/nix/parts/all.nix:29]: I'm resolving the conversation. Please open a PR when this one is merged

File: tools/nix/parts/all.nix (hunk 2 above)
Comment (etorreborre) [reply tools/nix/parts/all.nix:27]: I'm resolving the conversation. Please open a PR when this one is merged

**Review by davide-baldo — APPROVED** [review]

**Pull request merged**